temporal-sdk/src/main/java/io/temporal/internal/common/SearchAttributePayloadConverter.java

```diff
@@ -46,6 +47,84 @@ final class SearchAttributePayloadConverter {
   public static final SearchAttributePayloadConverter INSTANCE =
       new SearchAttributePayloadConverter();
 
+  public Payload encodeTyped(SearchAttributeKey<?> key, @Nullable Object value) {
+    // We can encode as-is because we know it's strictly typed to expected key value. We
+    // accept a null value because updates for unset can be null.
+    return DefaultDataConverter.STANDARD_INSTANCE.toPayload(value).get().toBuilder()
+        .putMetadata(
+            METADATA_TYPE_KEY,
+            ByteString.copyFromUtf8(indexValueTypeToEncodedValue(key.getValueType())))
+        .build();
+  }
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  public void decodeTyped(SearchAttributes.Builder builder, String name, @Nonnull Payload payload) {
+    // Get key type
+    SearchAttributeKey key;
+    IndexedValueType indexType = getIndexType(payload.getMetadataMap().get(METADATA_TYPE_KEY));
+    switch (indexType) {
+      case INDEXED_VALUE_TYPE_TEXT:
+        key = SearchAttributeKey.forText(name);
+        break;
+      case INDEXED_VALUE_TYPE_KEYWORD:
+        key = SearchAttributeKey.forKeyword(name);
+        break;
+      case INDEXED_VALUE_TYPE_INT:
+        key = SearchAttributeKey.forLong(name);
+        break;
+      case INDEXED_VALUE_TYPE_DOUBLE:
+        key = SearchAttributeKey.forDouble(name);
+        break;
+      case INDEXED_VALUE_TYPE_BOOL:
+        key = SearchAttributeKey.forBoolean(name);
+        break;
+      case INDEXED_VALUE_TYPE_DATETIME:
+        key = SearchAttributeKey.forOffsetDateTime(name);
+        break;
+      case INDEXED_VALUE_TYPE_KEYWORD_LIST:
```

Comment:
Before this PR `INDEXED_VALUE_TYPE_KEYWORD_LIST` didn't work in Java. I am not totally confident this fixes it. Can we add tests for the list cases? Maybe you did add them and I just missed them

Reply:
Yes, I need to add those for sure

Reply:
Added tests here. You were correct, there was a problem decoding side on Java test server which was fixed.